### PR TITLE
fix(app): ODD fixture removal, mag block with staging in deck map, add modules to drop tip wizard

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

Fix three bugs associated with the addition of modules to deck configuration. First repair the ODD's
deck configurator such that local changes are staged properly. Second, properly display the mag
block when loaded into a staging area fixture on the base deck component. Lastly, fix drop tip
wizard in the case of a configured module.

Closes [RQA-2622](https://opentrons.atlassian.net/browse/RQA-2622), Closes [RQA-2638](https://opentrons.atlassian.net/browse/RQA-2638)

# Review Requests

- configure and unconfigure fixtures from the ODD deck configurator successfully
- load a mag block into a column 3 slot in a protocol that also uses the adjacent column 4 staging area slot
- run drop tip wizard successfully and choose a location to drop in that has a module configured in it

# Risk assessment
low

[RQA-2622]: https://opentrons.atlassian.net/browse/RQA-2622?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RQA-2638]: https://opentrons.atlassian.net/browse/RQA-2638?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: components/src/hardware-sim/BaseDeck/BaseDeck.tsx [hunk 1]
```diff
@@ -114,7 +115,9 @@ export function BaseDeck(props: BaseDeckProps): JSX.Element {
   )
   const stagingAreaFixtures = deckConfig.filter(
     fixture =>
-      fixture.cutoutFixtureId === STAGING_AREA_RIGHT_SLOT_FIXTURE &&
+      (fixture.cutoutFixtureId === STAGING_AREA_RIGHT_SLOT_FIXTURE ||
+        fixture.cutoutFixtureId ===
+          STAGING_AREA_SLOT_WITH_MAGNETIC_BLOCK_V1_FIXTURE) &&
```
Comment (brenthagen) [inline components/src/hardware-sim/BaseDeck/BaseDeck.tsx:120]: for consistency we could do a constant `STAGING_AREA_FIXTURES` and a `STAGING_AREA_FIXTURES.includes(fixture.cutoutFixtureId)` filter check like we do for the other fixture types

Comment (vegano1) [reply components/src/hardware-sim/BaseDeck/BaseDeck.tsx:120]: I'll add a comment so we can come back to this, not too familiar with the stating areas vs cutouts.

**Review by brenthagen — APPROVED** [review]
changes make sense to me !

File: components/src/hardware-sim/BaseDeck/BaseDeck.tsx (hunk 1 above)
Comment (brenthagen) [reply components/src/hardware-sim/BaseDeck/BaseDeck.tsx:120]: yeah not urgent

**Pull request merged**